### ISSUE TITLE
[Feature request]:  add --with-mps build option

### ISSUE THREAD

**Issue by nailuoGG** [d12frosted/homebrew-emacs-plus#706]
### Feature request description

If we make it easier for more people to use the `scratch/cgi` [branch](https://github.com/emacs-mirror/emacs/tree/scratch/igc) with MPS support, they may be able to test it and provide feedback, which could help that branch develop further.

I've tried modifying the `emacs-plus` formula code to work with `scratch/cgi` branch , and I was able to successfully build a product that can handle daily work, but it still doesn't support some build options.

The options that cannot be compiled (now) are: `--with-xwidgets` and `--with-poll`

The command that can be compiled is:

```bash
brew install emacs-plus@31 --with-imagemagick --with-native-comp --with-modern-alecive-flatwoken-icon
```

Note: This build still occasionally crashes.

**Comment by d12frosted** [comment]
Is there any TLDR of what that branch is about? 

> The options that cannot be compiled (now) are: --with-xwidgets and --with-poll

I think at some point XWidgets Support will be dropped. It's experimental and AFAIK no that many people use it.

> Note: This build still occasionally crashes.

I wonder why.

**Comment by nailuoGG** [comment]
> Is there any TLDR of what that branch is about?

TL;DR: Replace Emacs' built-in GC with a [better one](https://github.com/Ravenbrook/mps). 

Progress: 
- [STatus of MPS branch](https://mail.gnu.org/archive/html/emacs-devel/2024-04/msg00434.html)
- [MPS: master merged](https://lists.gnu.org/archive/html/emacs-devel/2024-08/msg00111.html) Note: merged `master` into `scratch/igc`




> The options that cannot be compiled (now) are: --with-xwidgets and --with-poll

 The build process fails when combining -`-with-xwidgets` or `--with-poll` with `--with-mps`.  
 
>  Note: This build still occasionally crashes.

Error logs were not captured or saved. 

I was using the igc branch built with --with-mps approximately three days before opening this issue.

**Comment by d12frosted** [comment]
We had some discussion in #707. As result, let's see how development of this feature evolves and we can reconsider adding this option later. But for now, I am closing this as there are no plans to add this option.
